File: server/src/GoldFieldsHR.Domain/Entities/Acknowledgment.cs
using GoldFieldsHR.Domain.Enums;

namespace GoldFieldsHR.Domain.Entities;

public class Acknowledgment
{
    public Guid Id { get; set; }

    public AcknowledgmentEntityType EntityType { get; set; }
    public Guid EntityId { get; set; }

    public Guid EmployeeId { get; set; }
    public Employee? Employee { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

File: client/src/types/acknowledgment.ts
export const AcknowledgmentEntityType = {
  PpeRequest: 2,
  LegalAppointment: 3,
} as const;
export type AcknowledgmentEntityType = (typeof AcknowledgmentEntityType)[keyof typeof AcknowledgmentEntityType];

export interface AcknowledgmentDto {
  id: string;
  entityType: AcknowledgmentEntityType;
  entityId: string;
  employeeId: string;
  employeeName: string;
  createdAtUtc: string;
}
